src/tir/analysis/deep_equal.cc

```diff
@@ -70,10 +384,45 @@ bool ExprDeepEqual::operator()(const PrimExpr& lhs, const PrimExpr& rhs) const {
   return DeepCmpSEqualHandler().SEqualReduce(lhs, rhs, false, NullOpt);
 }
 
+class CommutativeDeepEqual : public ExprDeepEqual {
+ public:
+  bool operator()(const PrimExpr& lhs, const PrimExpr& rhs) const {
+    // quick path
+    if (lhs.same_as(rhs)) return true;
+    if (!lhs.defined() && rhs.defined()) return false;
+    if (!rhs.defined() && lhs.defined()) return false;
+    if (lhs->type_index() != rhs->type_index()) return false;
+    if (auto* plhs = lhs.as<IntImmNode>()) {
+      auto* prhs = rhs.as<IntImmNode>();
+      return plhs->dtype == prhs->dtype && plhs->value == prhs->value;
+    }
+    if (lhs.as<AnyNode>()) {
+      return false;
+    }
+    SortExprByHashMutator sort;
+    sort.pre_max_tree_idx = INT32_MAX;
+    auto sort_lhs = sort.Rewrite(lhs);
+    while (sort.pre_max_tree_idx != -1) {
+      sort_lhs = sort.Rewrite(sort_lhs);
+    }
+    sort.pre_max_tree_idx = INT32_MAX;
+    auto sort_rhs = sort.Rewrite(rhs);
```

Comment:
Replace with `SortExprByHashMutator::Rewrite(...)` and avoid using the same sorter twice.